llrt_core/src/modules/crypto/mod.rs

```diff
@@ -132,6 +133,49 @@ fn random_fill_sync<'js>(
     Ok(obj)
 }
 
+macro_rules! fill_typed_array {
+    ($ty:ty, $bytes:expr, $rng:expr) => {{
+        let size = mem::size_of::<$ty>();
+        for chunk in $bytes.chunks_exact_mut(size) {
+            let val_bytes = $rng.gen::<$ty>().to_ne_bytes();
+            chunk.copy_from_slice(&val_bytes[..size]);
+        }
+    }};
+}
+
+fn get_random_values<'js>(ctx: Ctx<'js>, obj: Object<'js>) -> Result<Object<'js>> {
+    let mut rng = rand::thread_rng();
+
+    if let Some(array_buffer) = obj_to_array_buffer(&ctx, &obj)? {
```

Comment:
I would suggest to refactor this method:
https://github.com/nabetti1720/llrt/blob/724f74f198fb93f3f674810d1728e203a4e8f75c/llrt_core/src/utils/object.rs#L142 to avoid matching on class names and instead to `if let Ok(array) = TypedArray::<i8>::from_object(obj.clone())` for each type and return the buffer.

Then use the same principle below.

Reply:
Is the correction as expected? Please review. :)